tests/Asset/BuildFileLocatorTest.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/*
+ * This file is part of the Symfony WebpackEncoreBundle package.
+ * (c) Fabien Potencier <fabien@symfony.com>
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\WebpackEncoreBundle\Tests\Asset;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\WebpackEncoreBundle\Asset\BuildFileLocator;
+
+class BuildFileLocatorTest extends TestCase
+{
+    /**
+     * @dataProvider getFindFileTests
+     */
+    public function testFindFile(string $buildPath, string $assetPath, string $expectedPath)
+    {
+        $fileLocator = new BuildFileLocator(['_default' => $buildPath]);
+        $fileLocator->disableFileExistsCheck();
+
+        $this->assertSame($expectedPath, $fileLocator->findFile($assetPath));
+    }
+
+    public function getFindFileTests()
+    {
+        yield 'no_overlap' => [
+            '/app/public',
+            'foo.js',
+            '/app/public/foo.js'
+        ];
+
+        yield 'simple_overlap' => [
+            '/app/public/build',
+            'build/foo.js',
+            '/app/public/build/foo.js'
+        ];
+
+        yield 'simple_overlap_with_slash' => [
+            '/app/public/build',
+            '/build/foo.js',
+            '/app/public/build/foo.js'
+        ];
+
+        yield 'simple_overlap_with_build_path_slash' => [
+            '/app/public/build/',
+            'build/foo.js',
+            '/app/public/build/foo.js'
+        ];
+
+        yield 'partial_overlap' => [
+            '/app/public/build',
+            'build/subdirectory/foo.js',
+            '/app/public/build/subdirectory/foo.js'
+        ];
+
+        yield 'overlap_in_wrong_spot' => [
+            '/app/public/build',
+            'subdirectory/build/foo.js',
+            '/app/public/build/subdirectory/build/foo.js'
+        ];
+
+        yield 'windows_paths' => [
+            'C:\\\\app\\public\\build',
+            'build/foo.js',
+            'C:\\\\app\\public\\build/build/foo.js'
+        ];
```

Comment:
Are files from dev-server supported?

Because that's an issue my team and me faced when working with Encore dev-server and assets in Twig for emails (with `inky_to_html` and `inline_css`

We were able to make them work with a custom extension:

```php
class EncoreExtension extends AbstractExtension
{
    private $publicDir;

    public function __construct(string $publicDir)
    {
        $this->publicDir = $publicDir;
    }

    public function getFunctions(): array
    {
        return [
            new TwigFunction('encore_source', [$this, 'source']),
        ];
    }

    public function source(string $path): string
    {
        // don't do anything if $path is http://my-app.domain:8008/...
        if (0 === strpos($path, '/')) {
            $path = $this->publicDir.$path;
        }

        // fetch content from filesystem or from http
        if (false !== $content = file_get_contents($path)) {
            return $content;
        }

        throw new \RuntimeException("Unable to read file \"$path\".");
    }
}
```

```twig
{% set css_to_inline = encore_entry_css_files('email') | map(file => encore_source(file)) | join('') %}
{% set css_to_ship = encore_entry_css_files('email-responsive') | map(file => encore_source(file)) | join('') %}

{% apply inky_to_html|inline_css(css_to_inline) %}
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width">

    <style>
    {{ css_to_ship }}
    </style>

    {# ... #}
```

But that would be really nice if it can be supported _natively_ here :D

Reply:
That's a really good point - I hadn't forgotten about the dev server. Assuming it *did* support the dev server (it actually does not currently), would the new `encore_entry_css_source()` Twig function work well for you?

Reply:
I will give it a try!

Reply:
I've updated my code to this:
```diff
-{% set css_to_inline = encore_entry_css_files('email') | map(file => encore_source(file)) | join('') %}
-{% set css_to_ship = encore_entry_css_files('email-responsive') | map(file => encore_source(file)) | join('') %}
- 
-{% apply inky_to_html|inline_css(css_to_inline) %}
+{% apply inky_to_html|inline_css(encore_entry_css_source('email')) %}
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width">

    <style>
-    {{ css_to_ship }}
+    {{ encore_entry_css_source('email-responsive') }}
    </style>

    <style>
```

It works when running `encore dev`, but not when running the dev-server:

![Capture d’écran de 2020-05-09 00-42-17](https://user-images.githubusercontent.com/2103975/81454975-04322480-918e-11ea-8a56-a5269cc36f88.png)